platform/util/platform-util/src/test/java/org/codice/ddf/platform/util/properties/PropertiesLoaderTest.java

```diff
@@ -0,0 +1,200 @@
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.util.properties;
+
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableMap;
+import java.io.File;
+import java.nio.file.Files;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PropertiesLoaderTest {
+
+  @Mock private Properties propertiesMock;
+
+  @ClassRule public static TemporaryFolder temporaryFolder = new TemporaryFolder();
+  private static final String PROPERTIES_FILENAME = "test.properties";
+  private static final String NON_EXISTENT_FILENAME = "nonExistentFile.properties";
+  private static File propertiesFile;
+
+  private static Map<Object, Object> exampleMap = ImmutableMap.of("a", "A", "b", "B", "c", "C");
+
+  private static final PropertiesLoader PROPERTIES_LOADER = PropertiesLoader.getInstance();
+
+  @BeforeClass
+  public static void setUpClass() throws Exception {
+
+    // set up string to put in properties file
+    StringBuilder sb = new StringBuilder();
+    for (Map.Entry<Object, Object> entry : exampleMap.entrySet()) {
+      sb.append(entry.getKey() + "=" + entry.getValue() + "\n");
+    }
+
+    // initialize file
+    propertiesFile = temporaryFolder.newFile(PROPERTIES_FILENAME);
+    Files.write(propertiesFile.toPath(), sb.toString().getBytes());
+  }
+
+  @Test
+  public void testToMapWithProperties() throws Exception {
+    when(propertiesMock.entrySet()).thenReturn(exampleMap.entrySet());
+
+    Map<?, ?> entries = PROPERTIES_LOADER.toMap(propertiesMock);
+    Assert.assertEquals("Returns a map of the properties", exampleMap, entries);
+  }
+
+  @Test
+  public void testToMapWithNull() throws Exception {
+    Map<?, ?> entries = PROPERTIES_LOADER.toMap(null);
+    Assert.assertEquals("Returns an empty map", Collections.emptyMap(), entries);
+  }
+
+  @Test
+  public void testToMapWithEmptyProperties() throws Exception {
+    Map<?, ?> entries = PROPERTIES_LOADER.toMap(new Properties());
+    Assert.assertEquals("Returns an empty map", Collections.emptyMap(), entries);
+  }
+
+  @Test
+  public void testLoadPropertiesWithoutPropertiesFile() throws Exception {
+    Properties testProperties = PROPERTIES_LOADER.loadProperties(null, null);
+    Properties emptyProperties = new Properties();
+
+    Assert.assertEquals("Returns empty properties", emptyProperties, testProperties);
+  }
+
+  @Test
+  public void testLoadPropertiesWithEmptyPropertiesFile() throws Exception {
+    File emptyFile = temporaryFolder.newFile("tempFile");
```

Comment:
✏️  Can we move this to the `@BeforeClass` so all setup and initialization occurs in the same place?

Reply:
I would rather keep this in here because this is the only test that uses an extra temporary file. If I was making extra files for all the tests I would see the benefit of moving those initializations into the before class because then that would be the spot where temporary files are made. But If only 1 test is making an extra file I think it makes more sense to put that logic in that test rather than pulling it out into its own section of 1 extra initialized file.